packages/material-ui/src/utils/useControlled.js

```diff
@@ -0,0 +1,44 @@
+/* eslint-disable react-hooks/rules-of-hooks, react-hooks/exhaustive-deps */
+import React from 'react';
+
+const useControlled = ({ controlled, default: defaultProp, name }) => {
```

Comment:
We will likely need to move it to the utils package (so we can use it in the lab too).